18F/identity-design-system
# Improve browser-ready bundle compatibility

## 🛠 Summary of changes

Improves a number of issues and inconsistencies with the prebuilt JavaScript bundle:

- Adds support for an equivalent to USWDS's `uswds-init.js` script to reduce flashing of content
- Fixes issue where the prebuilt bundle was not downgraded for target browser support
- Minifies the prebuilt JavaScript bundle
- Adds usage documentation to README.md

These issues were surfaced as part of ongoing work in https://github.com/18F/identity-dashboard/pull/748

## 📜 Testing Plan

Read included README.md and verify the usage instructions are easy to follow.

Verify test passes for initializer behavior:

1. `make build`
2. `node --test test/init.test.js`

Verify that `dist/` assets are minified:

1. `make build:js`
2. Open `dist/assets/js/init.js` and `dist/assets/js/main.js`
3. Observe that contents are minified

Verify that `init.js` is included in all output variants:

1. `make build`
2. Find `init.js` in: `dist/assets/js`, `build/cjs`, `build/esm`
